StrongInject.Tests.Unit/OwnedInjectionTests.cs

```diff
@@ -0,0 +1,1275 @@
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace StrongInject.Generator.Tests.Unit
+{
+    public class OwnedInjectionTests : TestBase
+    {
+        public OwnedInjectionTests(ITestOutputHelper outputHelper) : base(outputHelper)
+        {
+        }
+
+        // TODO: Investigate deduplicating when the same Owned<B> is injected twice, including deduplicating inside to outside of a func lambda
```

Comment:
> including deduplicating inside to outside of a func lambda

I wouldn't bother doing this unless you particularly want to - There's plenty of cases where this wont be allowed (for example if you use of of the delegate parameters in resolving the `Owned`) and it could significantly complexicate the code (I'm owning that word).

I have a feeling that within the same resolution scope this should just fall out automatically. Have you tested? If it doesn't work automatically, look at `_existingVariables` in `LoweringVisitor`.

Reply:
That's definitely wrong

Reply:
Fixed in latest push. Now I just need to come up with a way to reuse local function declarations.